File: Examples/BinaryExampleIPA/BadPracticesIPA/BadPracticesIPA/CacomiTestApp/Network/AuthSession.swift
//
//  AuthSession.swift
//  CacomiTestApp
//
//  Fixtures for AuthSessionRule.
//

import Foundation

enum AuthSession {

    // CACOMI-EXPECT[AuthSessionRule|high]: session never expires (TimeInterval.greatestFiniteMagnitude)
    static let sessionExpiry: TimeInterval = .greatestFiniteMagnitude

    static func loginURL(token: String) -> URL? {
        // CACOMI-EXPECT[AuthSessionRule|high]: bearer token placed in URL query parameter
        return URL(string: "https://api.fake-cacomi.com/session?token=\(token)")
    }

    static func persistSessionCookie(_ value: String) {
        // CACOMI-EXPECT[AuthSessionRule|high]: session cookie stored without HttpOnly/Secure flags
        let cookie = HTTPCookie(properties: [
            .name: "SESSIONID",
            .value: value,
            .domain: "fake-cacomi.com",
            .path: "/"
        ])
        if let cookie = cookie {
            HTTPCookieStorage.shared.setCookie(cookie)
        }
    }

    // CACOMI-NEGATIVE[AuthSessionRule]: token sent via Authorization header, session has expiry
    static func secureRequest(token: String) -> URLRequest {
        var req = URLRequest(url: URL(string: "https://api.fake-cacomi.com/me")!)
        req.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
        req.timeoutInterval = 30
        return req
    }
}

File: Examples/BinaryExampleAPK/BadPracticesAppAndroid/app/src/main/java/com/example/badpracticesappandroid/network/AuthSession.kt
package com.example.badpracticesappandroid.network

import android.webkit.CookieManager
import java.net.HttpCookie

/**
 * Equivalente Android de AuthSession.swift — AuthSessionRule.
 */
object AuthSession {

    // BAD: la sesion nunca expira // CACOMI-EXPECT: AuthSessionRule
    const val SESSION_EXPIRY_MILLIS: Long = Long.MAX_VALUE

    // BAD: bearer token en query param de la URL (se filtra en logs/proxies) // CACOMI-EXPECT: AuthSessionRule
    fun loginUrl(token: String): String =
        "https://api.fake-cacomi.com/session?token=$token"

    // BAD: cookie de sesion sin flags HttpOnly/Secure // CACOMI-EXPECT: AuthSessionRule
    fun persistSessionCookie(value: String) {
        val cookie = HttpCookie("SESSIONID", value).apply {
            domain = "fake-cacomi.com"
            path = "/"
            // sin secure, sin httpOnly
        }
        CookieManager.getInstance().setCookie("https://fake-cacomi.com", cookie.toString())
    }

    // NEGATIVO: token en header Authorization + timeout/expiry definido // CACOMI-EXPECT: none
    fun secureRequestHeaders(token: String): Map<String, String> = mapOf(
        "Authorization" to "Bearer $token"
    )
}
